core/src/main/java/org/apache/iceberg/avro/AvroSchemaVisitor.java

```diff
@@ -52,8 +52,21 @@
       case UNION:
         List<Schema> types = schema.getTypes();
         List<T> options = Lists.newArrayListWithExpectedSize(types.size());
-        for (Schema type : types) {
-          options.add(visit(type, visitor));
+        if (AvroSchemaUtil.isOptionSchema(schema)) {
```

Comment:
can you please verify that current definition of `isOptionSchema`, where it checks for null option in specific order/index, is still applicable here?

Reply:
I'm not quite following this. I use `isOptionSchema` for the vanilla union case where it is composed of only 2 types one is null and one is the other type `[null, someType]` .